hazelcast/src/test/java/com/hazelcast/client/tpc/ClientAsyncSocketReaderTest.java

```diff
@@ -0,0 +1,259 @@
+            ClientTestUtil.writeClientMessage(os, request);
+            assertSocketClosed(is);
+        }
+    }
+
+    @Test
+    public void testMessageSizeLimits_afterAuth() throws IOException {
+        HazelcastInstance server = newServer();
+        HazelcastInstance client = newClient();
+
+        int port = getTpcPort(server);
+        try (Socket socket = new Socket("127.0.0.1", port);
+             OutputStream os = socket.getOutputStream();
+             InputStream is = socket.getInputStream()) {
+
+            os.write(Protocols.CLIENT_BINARY.getBytes(StandardCharsets.UTF_8));
+            ClientMessage request = encodeAuthRequest(getUuid(client), getTpcToken(server));
+            ClientTestUtil.writeClientMessage(os, request);
+            ClientMessage response = ClientTestUtil.readResponse(is);
+            assertEquals(ExperimentalTpcAuthenticationCodec.RESPONSE_MESSAGE_TYPE, response.getMessageType());
+
+            ClientMessage largeRequest = encodeAuthRequest(getUuid(client), new byte[UNVERIFIED_MESSAGE_LENGTH_LIMIT + 1]);
+            ClientTestUtil.writeClientMessage(os, largeRequest);
+            ClientMessage largeResponse = ClientTestUtil.readResponse(is);
+            assertEquals(ExperimentalTpcAuthenticationCodec.RESPONSE_MESSAGE_TYPE, largeResponse.getMessageType());
+        }
+    }
+
+    @Test
+    public void testMessagesBeforeAuth() throws IOException {
+        HazelcastInstance server = newServer();
+
+        int port = getTpcPort(server);
+        try (Socket socket = new Socket("127.0.0.1", port);
+             OutputStream os = socket.getOutputStream();
+             InputStream is = socket.getInputStream()) {
+
+            os.write(Protocols.CLIENT_BINARY.getBytes(StandardCharsets.UTF_8));
+            ClientMessage request = ClientPingCodec.encodeRequest();
+            ClientTestUtil.writeClientMessage(os, request);
+            assertSocketClosed(is);
+        }
+    }
+
+    @Test
+    public void testAuth_withNotFoundUuid() throws IOException {
+        HazelcastInstance server = newServer();
+        newClient();
+
+        int port = getTpcPort(server);
+        try (Socket socket = new Socket("127.0.0.1", port);
+             OutputStream os = socket.getOutputStream();
+             InputStream is = socket.getInputStream()) {
+
+            os.write(Protocols.CLIENT_BINARY.getBytes(StandardCharsets.UTF_8));
+            ClientMessage request = encodeAuthRequest(UUID.randomUUID(), getTpcToken(server));
+            ClientTestUtil.writeClientMessage(os, request);
+            assertSocketClosed(is);
+        }
+    }
+
+    @Test
+    public void testAuth_withMismatchedToken() throws IOException {
+        HazelcastInstance server = newServer();
+        HazelcastInstance client = newClient();
+
+        int port = getTpcPort(server);
+        try (Socket socket = new Socket("127.0.0.1", port);
```

Comment:
Is using `server.getCluster().getLocalMember().getAddress().getHost()` instead of "127.0.0.1" safer?

Reply:
https://github.com/hazelcast/hazelcast/pull/24236/commits/e6a3a1839ded53ddf2be59073b97dc629e61823e